e2e/stack.go

```diff
@@ -69,72 +64,69 @@ func New(
 }
 
 func (s *Stack) Run(ctx context.Context, env *environment.Env) error {
-	// Run anvil.
-	anvilCmd := exec.CommandContext( //nolint:gosec
-		ctx,
-		"anvil",
-		"--port", s.anvilURL.Port(),
-		"--order", "fifo",
-		"--disable-block-gas-limit",
-		"--gas-price", "0",
-		"--block-time", fmt.Sprint(s.l1BlockTime.Seconds()),
-	)
-	anvilCmd.Cancel = func() error {
-		// Anvil can catch SIGTERMs. The exec package sends a SIGKILL by default.
-		return anvilCmd.Process.Signal(syscall.SIGTERM)
+	// configure & run L1
+
+	const l2ChainID = 901
+	const networkName = "hardhat"
+	l1Deployments, err := opgenesis.NewL1Deployments("optimism/.devnet/addresses.json")
+	if err != nil {
+		return fmt.Errorf("new l1 deployments: %v", err)
 	}
-	if err := s.startCmd(anvilCmd); err != nil {
-		return err
+	deployConfig, err := opgenesis.NewDeployConfigWithNetwork(networkName, filepath.Join(s.contractsRootDir, "deploy-config"))
```

Comment:
I think we should use the deploy config generated by `make setup-e2e`. (see [here](https://github.com/ethereum-optimism/optimism/blob/630cf349f253aaaf978db64e95c7ccc8e89059f8/op-e2e/config/init.go#L75) in op-e2e)

Reply:
:+1: 8427d1a

Reply:
I've confirmed that `devnetl1.json` gets generated by `make e2e`, but frustratingly it can't be consumed by `opgenesis.NewDeployConfigWithNetwork("devnetL1")`.

Fails with:

```
new deploy config: cannot unmarshal deploy config: json: unknown field "faultGameMaxDuration"
```

When I force the deploy constructor to unmarshall, the test hangs indefinitely.

Expecting now that this is a local go packaging issue. Will try running CI against the generated `devnetL1` network.